R/repl.R

```diff
@@ -37,6 +37,31 @@
 #'
 #' @seealso [py], for accessing objects created using the Python REPL.
 #'
+#' @section Magics: A handful of magics are supported in `repl_python()`:
+#'
+#' Lines prefixed with `!` are executed as system commands:
+#'  - `!cmd --arg1 --arg2`: Execute arbitrary system commands
+#'
+#' Magics start with a `%` prefix. Supported magics include:
+#'  - `%conda ...` executes a conda command in the active conda environment
+#'  - `%pip ...` executes pip for the active python.
+#'  - `%load`, `%loadpy`, `%run` executes a python file.
+#'  - `%system`, `!!` executes a system command and capture output
+#'  - `%env`: read current environment variables.
+#'    - `%env name`: read environment variable 'name'.
+#'    - `%env name=val`, `%env name val`: set environment variable 'name' to 'val'.
+#'       `val` elements in `{}` are interpolated using f-strings (required Python >= 3.6).
+#'  - `%cd <dir>` change working directory.
+#'     - `%cd -`: change to previous working directory.
+#'     - `%cd -3`: change to 3rd most recent working directory.
+#'     - `%cd -foo/bar`: change to most recent working directory matching `"foo/bar"` regex.
+#'  - `%pwd`: print current working directory.
+#'  - `%dhist`: print working directory history.
+#'
+#' Additionally, the output of magic commands can be captured in a variable, e.g.:
+#'  - `x = !ls`
```

Comment:
Something to document: what exactly is captured here? Both stdout and stderr? Just stdout? Do we split on newlines for the user, or just capture all output as-is?